extensions/knowwa/src/modules/NewDoc/NewDocForm.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  Action,
+  ActionPanel,
+  Clipboard,
+  Form,
+  getPreferenceValues,
+  showToast,
+  Toast,
+  useNavigation,
+} from "@raycast/api";
+import axios from "axios";
+import dayjs from "dayjs";
+import { apiUrl, baseUrl } from "../../constants/constants";
+import { Document } from "../../types/types";
+
+interface Preferences {
+  AccessToken: string;
+}
+
+export default function NewDocForm({ template }: { template: Document }) {
+  const preferences = getPreferenceValues<Preferences>();
+  const { pop } = useNavigation();
+
+  const submitForm = async (values: object) => {
+    const toast = await showToast({
+      style: Toast.Style.Animated,
+      title: "Generating your Document",
+    });
+    try {
+      const data = await axios.post(
+        `${apiUrl}/documents/new/${template.uuid}`,
+        values,
+        {
+          headers: {
+            Authorization: `Bearer ${preferences["AccessToken"]}`,
+            accept: "application/json ",
+          },
+        }
+      );
+
+      await Clipboard.copy(`${baseUrl}/view-doc/${data.data.document_id}`);
+
+      pop();
+      toast.style = Toast.Style.Success;
+      toast.title = "Document Generated and URL copied to clipboard!";
+    } catch (err) {
+      toast.style = Toast.Style.Failure;
+      toast.title = "Could not generate Document";
+    }
+  };
+
+  return (
+    <Form
```

Comment:
Are there some required fields? You might want to validate with [Validation](https://developers.raycast.com/api-reference/user-interface/form#validation)

Reply:
The API of Knowwa currently does not support marking required fields, but I will keep it on the radar for next releases. The good news is, the submission will never fail due to missing or wrongly formatted fields yet.